test/battle/move_effect/natural_gift.c

```diff
@@ -1,4 +1,66 @@
 #include "global.h"
 #include "test/battle.h"
 
+SINGLE_BATTLE_TEST("Natural Gift removes berry if move fails due to an immunity")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_PECHA_BERRY); }
+        OPPONENT(SPECIES_PHANPY);
+    } WHEN {
+        TURN { MOVE(player, MOVE_NATURAL_GIFT); }
+    } SCENE {
+        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_NATURAL_GIFT, player);
+    } THEN {
+        EXPECT(player->item == ITEM_NONE);
+    }
+}
+
+SINGLE_BATTLE_TEST("Natural Gift does not remove berry if user is ejected out")
+{
+    KNOWN_FAILING; // The test runner does not remember items given to a species.
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_PECHA_BERRY); }
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_RED_CARD); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_NATURAL_GIFT); }
+        TURN { SWITCH(player, 1); }
```

Comment:
```suggestion
        TURN { SWITCH(player, 0); }
```

Should fix this test (probably shouldn't allow a battler to switch to themselves in tests)

Reply:
fixed, thx!